packages/compass-collection/src/components/collection-tab.tsx

```diff
@@ -118,10 +118,10 @@ function WithErrorBoundary({
 function useCollectionTabs(props: CollectionMetadata) {
   const pluginTabs = useCollectionSubTabs();
   const connectionInfoRef = useConnectionInfoRef();
-  const isGlobalWritesSupported = useConnectionSupports(
-    connectionInfoRef.current.id,
-    'globalWrites'
-  );
+  const isGlobalWritesSupported =
+    useConnectionSupports(connectionInfoRef.current.id, 'globalWrites') &&
+    !props.isReadonly &&
+    !['config', 'local', 'admin'].includes(props.namespace.split('.')[0]);
```

Comment:
the logic is correct but I imagine there's a more maintainable location to put it. Suggestions welcome here.

Unfortunately I don't know the logic behind config/local/admin being not sharded -- this is just copying from data explorer code.